src/app/admin/collectionCentresTable/CollectionCentresTable.tsx

```diff
@@ -99,6 +173,111 @@ const CollectionCentresTable: React.FC = () => {
+        const { error: updateTimeSlotError } =
+            await updateDbCollectionCentreTimeSlots(timeSlotModalData);
+        const baseAuditLog = getBaseAuditLogForCollectionCentreTimeSlots(
+            "update collection centre time slots",
+            timeSlotModalData
+        );
+
+        if (updateTimeSlotError) {
+            setTimeSlotModalErrorMessage(
+                `Failed to update the collection centre time slots. Log ID: ${updateTimeSlotError.logId}`
+            );
+            await sendAuditLog({
+                ...baseAuditLog,
+                wasSuccess: false,
+                logId: updateTimeSlotError.logId,
+            });
+        }
+
+        await sendAuditLog({ ...baseAuditLog, wasSuccess: true });
+        setTimeSlotModalIsOpen(false);
+    };
+
+    const handleAddSlotClick = async (): Promise<void> => {
+        setEditableIsShown(true);
+    };
+
+    const checkIfSlotExists = (
+        existingTimeSlotData: FormattedTimeSlotsWithPrimaryKey,
+        newTimeSlot: FormattedTimeSlot
+    ): boolean => {
+        return existingTimeSlotData.timeSlots.some((slot) => slot.time === newTimeSlot.time);
+    };
+
+    const addNewTimeSlotToTimeSlotModalData = (
+        existingTimeSlotData: FormattedTimeSlotsWithPrimaryKey,
+        newTimeSlot: FormattedTimeSlot
+    ): void => {
+        const newTimeSlotArray = [...existingTimeSlotData.timeSlots, newTimeSlot];
+        newTimeSlotArray.sort((slot1, slot2) => slot1.time.localeCompare(slot2.time));
+
+        const updatedTimeSlotModalData: FormattedTimeSlotsWithPrimaryKey = {
+            ...existingTimeSlotData,
+            timeSlots: newTimeSlotArray,
+        };
+
+        setTimeSlotModalData(updatedTimeSlotModalData);
+    };
+
+    const handleSaveSlotClick = async (): Promise<void> => {
+        setTimeSlotModalErrorMessage(null);
+        if (timeSlotModalData === null || collectionTimeSlotValue === undefined) {
+            return;
+        }
+        const newTimeSlot: FormattedTimeSlot = {
+            time: formatDayjsToHoursAndMinutes(collectionTimeSlotValue),
+            isActive: false,
+        };
+
+        if (checkIfSlotExists(timeSlotModalData, newTimeSlot)) {
+            setTimeSlotModalErrorMessage(
+                "This time slot already exists. Please select a different time."
+            );
+            return;
+        }
+
+        addNewTimeSlotToTimeSlotModalData(timeSlotModalData, newTimeSlot);
+
+        setEditableIsShown(false);
+    };
+
+    const handleTimeSlotCheckBoxChange = (event: React.SyntheticEvent<Element, Event>): void => {
+        if (!timeSlotModalData) {
+            return;
+        }
+
+        const updatedTime = event.currentTarget.parentElement?.parentElement?.innerText;
+        const timeSlotIndex = timeSlotModalData.timeSlots.findIndex(
+            (slot) => slot.time === updatedTime
+        );
+        const timeSlot = timeSlotModalData.timeSlots.find((slot) => slot.time === updatedTime);
```

Comment:
I think if we have index we can just use that instead of searching again
```suggestion
        const timeSlot = timeSlotModalData.timeSlots[timeSlotIndex]
```